src/models/serialization.jl

```diff
@@ -1,98 +1,115 @@
+const _ENCODE_AS_UUID_A = (
+    Union{Nothing, Arc},
+    Union{Nothing, Area},
+    Union{Nothing, Bus},
+    Union{Nothing, LoadZone},
+    Union{Nothing, DynamicInjection},
+    Vector{Service},
+)
 
-# Enables deserialization of VariableCost. The default implementation can't figure out the
-# variable Union.
+const _ENCODE_AS_UUID_B = (Arc, Area, Bus, LoadZone, DynamicInjection, Vector{Service})
+@assert length(_ENCODE_AS_UUID_A) == length(_ENCODE_AS_UUID_B)
 
-function JSON2.read(io::IO, ::Type{VariableCost})
-    data = JSON2.read(io)
-    if data.cost isa Real
-        return VariableCost(Float64(data.cost))
-    elseif data.cost[1] isa Array
-        variable = Vector{Tuple{Float64, Float64}}()
-        for array in data.cost
-            push!(variable, Tuple{Float64, Float64}(array))
-        end
-    else
-        @assert data.cost isa Tuple || data.cost isa Array
-        variable = Tuple{Float64, Float64}(data.cost)
+should_encode_as_uuid(val) = mapreduce(x -> val isa x, |, _ENCODE_AS_UUID_B)
+should_encode_as_uuid(::Type{T}) where {T} = mapreduce(x -> T <: x, |, _ENCODE_AS_UUID_A)
```

Comment:
is this the same as `any` or `all`? e.g.`all(x -> T <: x,  _ENCODE_AS_UUID_A)`

Reply:
Yes, that would be better.  I'll change it.